sql/core/src/main/scala/org/apache/spark/sql/artifact/ArtifactManager.scala

```diff
@@ -72,6 +72,7 @@ class ArtifactManager(session: SparkSession) extends Logging {
 
   protected[artifact] val state: JobArtifactState =
     JobArtifactState(session.sessionUUID, Option(classURI))
+  session.jobArtifactState = state
```

Comment:
update spark session default jobArtifactState while create spark connect job artifact state
@HyukjinKwon is this way ok?